crates/engine/tree/src/tree/payload_processor/mod.rs

```diff
@@ -284,15 +285,16 @@ where
     }
 
     /// Spawn prewarming optionally wired to the multiproof task for target updates.
-    fn spawn_caching_with<P>(
+    fn spawn_caching_with<P, Tx>(
         &self,
         env: ExecutionEnv<Evm>,
-        mut transactions: mpsc::Receiver<impl ExecutableTxFor<Evm> + Send + 'static>,
+        mut transactions: mpsc::Receiver<Tx>,
         provider_builder: StateProviderBuilder<N, P>,
         to_multi_proof: Option<Sender<MultiProofMessage>>,
     ) -> CacheTaskHandle
     where
         P: BlockReader + StateProviderFactory + StateReader + Clone + 'static,
+        Tx: ExecutableTxFor<Evm> + Clone + Send + std::any::Any + 'static,
```

Comment:
This is required because in prewarm.rs, we use std::any::Any for runtime type checking in the check_is_deposit_tx function. Not sure if theres a better way to go about this? 

cc @mattsse